charmcraft/commands/pack.py

```diff
@@ -183,19 +174,28 @@ def _pack_bundle(self):
                 "file {!r}.".format(str(bundle_filepath))
             )
 
-        # so far 'pack' works for bundles only (later this will operate also on charms)
-        if self.config.type != "bundle":
-            raise CommandError(
-                "Bad config: 'type' field in charmcraft.yaml must be 'bundle' for this command."
-            )
+        # set prime filters
+        for fname in MANDATORY_FILES:
+            fpath = project.dirpath / fname
+            if not fpath.exists():
+                raise CommandError("Missing mandatory file: {!r}.".format(str(fpath)))
+        prime.extend(MANDATORY_FILES)
+
+        # set source for buiding
+        charm_part["source"] = str(project.dirpath)
+
+        # run the parts lifecycle
+        logger.debug("Parts definition: %s", config_parts)
+        lifecycle = PartsLifecycle(
+            config_parts,
+            work_dir=project.dirpath / build.BUILD_DIRNAME,
+            ignore_local_sources=[bundle_name + ".zip"],
```

Comment:
As above, do we need to consider `.jujuignore`?

Reply:
The current implementation of bundle creation doesn't parse `.jujuignore`. The ignore filter here prevents bundle files to include previous copies of themselves if they're created inside the bundle source directory.